keras_cv/layers/object_detection/non_max_suppression.py

```diff
@@ -90,13 +90,45 @@ def call(
 
         # TODO(tirthasheshpatel): Use backend-specific op where available
         if multi_backend():
-            idx, valid_det = non_max_suppression(
-                box_prediction,
-                confidence_prediction,
-                max_output_size=self.max_detections,
-                iou_threshold=self.iou_threshold,
-                score_threshold=self.confidence_threshold,
-            )
+            if keras.backend.backend() == "torch":
+                import torchvision
+
+                batch_size = box_prediction.shape[0]
+                idx = ops.zeros((batch_size, self.max_detections))
+                valid_det = ops.zeros((batch_size), "int32")
+
+                for batch_idx in range(batch_size):
+                    conf_mask = (
+                        confidence_prediction[batch_idx]
+                        > self.confidence_threshold
+                    )
+                    conf_mask_idx = ops.squeeze(ops.nonzero(conf_mask), axis=0)
+                    conf_i = confidence_prediction[batch_idx][conf_mask]
+                    box_i = box_prediction[batch_idx][conf_mask]
+
+                    idx_i = torchvision.ops.nms(
```

Comment:
Is this mainly for the efficiency of the torchvision op? If so, might be nice to leave a comment at the top of this block to clarify.

Reply:
indeed -- good call! I added a comment